File: app/expo-segmented-audio/android/src/main/java/expo/modules/segmentedaudio/SegmentedRegistry.kt
package expo.modules.segmentedaudio

import java.util.concurrent.ConcurrentHashMap

object SegmentedRegistry {
  data class Entry(val path: String, val key: ByteArray, val segmentSize: Int, val mimeType: String?)
  private val map = ConcurrentHashMap<String, Entry>()
  fun put(token: String, entry: Entry) {
    map[token] = entry
  }
  fun get(token: String): Entry? = map[token]
  fun remove(token: String) { map.remove(token) }
  fun clear() { map.clear() }
}

File: app/expo-segmented-audio/ios/SegmentedRegistry.swift
import Foundation

final class SegmentedRegistry {
  struct Entry {
    let path: String
    let keyData: Data
    let segmentSize: Int
    let mimeType: String?
  }

  static let shared = SegmentedRegistry()

  private let queue = DispatchQueue(label: "ExpoSegmentedAudio.SegmentedRegistry", attributes: .concurrent)
  private var map: [String: Entry] = [:]

  func put(token: String, entry: Entry) {
    queue.async(flags: .barrier) { [weak self] in
      self?.map[token] = entry
    }
  }

  func get(token: String) -> Entry? {
    queue.sync { map[token] }
  }

  func remove(token: String) {
    queue.async(flags: .barrier) { [weak self] in
      self?.map.removeValue(forKey: token)
    }
  }

  func clear() {
    queue.async(flags: .barrier) { [weak self] in
      self?.map.removeAll()
    }
  }
}


